internal/librarian/flags_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package librarian
+
+import (
+	"errors"
+	"flag"
+	"os"
+	"testing"
+)
+
+func TestValidatePush(t *testing.T) {
+	tests := []struct {
+		name           string
+		setFlagPush    bool
+		setGitHubToken bool
+		expectedErr    error
+	}{
+		{
+			name:           "push is false, no error",
+			setFlagPush:    false,
+			setGitHubToken: false,
+			expectedErr:    nil,
+		},
+		{
+			name:           "push is true, token exists, no error",
+			setFlagPush:    true,
+			setGitHubToken: true,
+			expectedErr:    nil,
+		},
+		{
+			name:           "push is true, no token, error",
+			setFlagPush:    true,
+			setGitHubToken: false,
+			expectedErr:    errors.New("no GitHub token supplied for push"),
+		},
+		{
+			name:           "push is false, no token, no error",
+			setFlagPush:    false,
+			setGitHubToken: true,
+			expectedErr:    nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resetAllFlags()
+
+			// Set up global flag and environment variable based on test case
```

Comment:
we should clean this up as part of https://github.com/googleapis/librarian/issues/198